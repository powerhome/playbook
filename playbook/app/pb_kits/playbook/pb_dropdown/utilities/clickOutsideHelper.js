export const handleClickOutside = ({ inputWrapperRef, dropdownContainerRef, setIsDropDownClosed, setFocusedOptionIndex, setIsInputFocused, }) => (e) => {
    let targetElement = e.target;
    let shouldClose = true;
    //Only needed for when useDropdown hook used with external trigger
    while (targetElement && shouldClose) {
        if (targetElement.getAttribute("data-dropdown") === "pb-dropdown-trigger") {
            shouldClose = false;
        }
        targetElement = targetElement.parentElement;
    }
    if (inputWrapperRef.current &&
        !inputWrapperRef.current.contains(e.target) &&
        dropdownContainerRef.current &&
        !dropdownContainerRef.current.contains(e.target) &&
        shouldClose) {
        setIsDropDownClosed(true);
        setFocusedOptionIndex(-1);
        setIsInputFocused(false);
    }
};
//# sourceMappingURL=clickOutsideHelper.js.map