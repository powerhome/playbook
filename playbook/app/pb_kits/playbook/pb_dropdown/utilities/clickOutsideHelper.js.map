{"version":3,"file":"clickOutsideHelper.js","sourceRoot":"","sources":["clickOutsideHelper.tsx"],"names":[],"mappings":"AAQA,MAAM,CAAC,MAAM,kBAAkB,GAC7B,CAAC,EACC,eAAe,EACf,oBAAoB,EACpB,mBAAmB,EACnB,qBAAqB,EACrB,iBAAiB,GACM,EAAE,EAAE,CAC7B,CAAC,CAAa,EAAE,EAAE;IAChB,IAAI,aAAa,GAAG,CAAC,CAAC,MAAqB,CAAC;IAC5C,IAAI,WAAW,GAAG,IAAI,CAAC;IAEvB,kEAAkE;IAClE,OAAO,aAAa,IAAI,WAAW,EAAE,CAAC;QACpC,IACE,aAAa,CAAC,YAAY,CAAC,eAAe,CAAC,KAAK,qBAAqB,EACrE,CAAC;YACD,WAAW,GAAG,KAAK,CAAC;QACtB,CAAC;QACD,aAAa,GAAG,aAAa,CAAC,aAA4B,CAAC;IAC7D,CAAC;IACD,IACE,eAAe,CAAC,OAAO;QACvB,CAAC,eAAe,CAAC,OAAO,CAAC,QAAQ,CAAE,CAAC,CAAC,MAAsB,CAAC;QAC5D,oBAAoB,CAAC,OAAO;QAC5B,CAAC,oBAAoB,CAAC,OAAO,CAAC,QAAQ,CAAE,CAAC,CAAC,MAAsB,CAAC;QACjE,WAAW,EACX,CAAC;QACD,mBAAmB,CAAC,IAAI,CAAC,CAAC;QAC1B,qBAAqB,CAAC,CAAC,CAAC,CAAC,CAAC;QAC1B,iBAAiB,CAAC,KAAK,CAAC,CAAC;IAC3B,CAAC;AACH,CAAC,CAAC"}