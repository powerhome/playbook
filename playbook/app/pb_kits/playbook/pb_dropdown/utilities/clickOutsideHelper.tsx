type HandleClickOutsideType = {
  inputWrapperRef?: React.RefObject<HTMLDivElement>;
  dropdownContainerRef?: React.RefObject<HTMLDivElement>;
  setIsDropDownClosed?: (value: boolean) => void;
  setFocusedOptionIndex?: (value: number) => void;
  setIsInputFocused?: (value: boolean) => void;
};

export const handleClickOutside =
  ({
    inputWrapperRef,
    dropdownContainerRef,
    setIsDropDownClosed,
    setFocusedOptionIndex,
    setIsInputFocused,
  }: HandleClickOutsideType) =>
  (e: MouseEvent) => {
    let targetElement = e.target as HTMLElement;
    let shouldClose = true;

    //Only needed for when useDropdown hook used with external trigger
    while (targetElement && shouldClose) {
      if (
        targetElement.getAttribute("data-dropdown") === "pb-dropdown-trigger"
      ) {
        shouldClose = false;
      }
      targetElement = targetElement.parentElement as HTMLElement;
    }
    if (
      inputWrapperRef.current &&
      !inputWrapperRef.current.contains((e.target as HTMLElement)) &&
      dropdownContainerRef.current &&
      !dropdownContainerRef.current.contains((e.target as HTMLElement)) &&
      shouldClose
    ) {
      setIsDropDownClosed(true);
      setFocusedOptionIndex(-1);
      setIsInputFocused(false);
    }
  };
