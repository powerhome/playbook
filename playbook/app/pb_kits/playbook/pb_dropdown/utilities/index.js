export { separateChildComponents, prepareSubcomponents } from './subComponentHelper';
export { handleClickOutside } from './clickOutsideHelper';
//# sourceMappingURL=index.js.map