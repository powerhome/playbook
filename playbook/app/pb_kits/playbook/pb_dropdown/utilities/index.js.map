{"version":3,"file":"index.js","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,uBAAuB,EAAE,oBAAoB,EAAE,MAAM,sBAAsB,CAAC;AACrF,OAAO,EAAE,kBAAkB,EAAE,MAAM,sBAAsB,CAAC"}