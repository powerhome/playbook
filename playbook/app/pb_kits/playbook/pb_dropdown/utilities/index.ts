export { separateChildComponents, prepareSubcomponents } from './subComponentHelper';
export { handleClickOutside } from './clickOutsideHelper';