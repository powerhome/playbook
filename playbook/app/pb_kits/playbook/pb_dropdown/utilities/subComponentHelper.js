import { jsx as _jsx } from "react/jsx-runtime";
import React from "react";
import DropdownTrigger from "../subcomponents/DropdownTrigger";
import DropdownContainer from "../subcomponents/DropdownContainer";
export const separateChildComponents = (children) => {
    let trigger = null;
    let container = null;
    const otherChildren = [];
    React.Children.forEach(children, (child) => {
        if (child && child.type === DropdownTrigger) {
            trigger = child;
        }
        else if (child && child.type === DropdownContainer) {
            container = child;
        }
        else {
            otherChildren.push(child);
        }
    });
    return { trigger, container, otherChildren };
};
export const prepareSubcomponents = ({ children, hasTriggerSubcomponent, hasContainerSubcomponent, trigger, container, otherChildren, dark }) => {
    const componentsToRender = [];
    if (!hasTriggerSubcomponent && !hasContainerSubcomponent) {
        componentsToRender.push(_jsx(DropdownTrigger, { dark: dark }));
        componentsToRender.push(_jsx(DropdownContainer, { dark: dark, children: children }));
    }
    else if (!hasTriggerSubcomponent && hasContainerSubcomponent) {
        componentsToRender.push(_jsx(DropdownTrigger, { dark: dark }));
        componentsToRender.push(children);
    }
    else if (hasTriggerSubcomponent && !hasContainerSubcomponent) {
        componentsToRender.push(trigger);
        componentsToRender.push(_jsx(DropdownContainer, { dark: dark, children: otherChildren }));
    }
    else {
        componentsToRender.push(trigger);
        componentsToRender.push(container);
    }
    return componentsToRender;
};
//# sourceMappingURL=subComponentHelper.js.map