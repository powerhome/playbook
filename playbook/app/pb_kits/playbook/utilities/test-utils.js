/* This file provides the base for tests */

import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

/*
  We can complicate this wrapper as needed.
  https://testing-library.com/docs/react-testing-library/setup
*/
// eslint-disable-next-line react/prop-types
const TestApp = ({ children }) => {
  return (
    <>
      { children }
    </>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: TestApp, ...options })

export * from '@testing-library/react'
export { customRender as render }
