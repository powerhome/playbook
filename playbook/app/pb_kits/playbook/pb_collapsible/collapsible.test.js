import React from 'react'
import { render, screen } from '../utilities/test-utils'

import Collapsible from './_collapsible'

beforeEach(() => {
  // Silences error logs within the test suite.
  jest.spyOn(console, 'error')
  // eslint-disable-next-line
  console.error.mockImplementation(() => {})
})

afterEach(() => {
  // eslint-disable-next-line
  console.error.mockRestore()
})

const testId = 'collapsible1',
  kitClass = 'pb_collapsible_kit'

test('throws error if incorrect # of children', () => {
  expect(() => {
    render(
      <Collapsible />
    )
  }).toThrow()
})

test('returns namespaced additional_class class name', () => {
  render(
    <Collapsible
        className="additional_class"
        data={{ testid: testId }}
    >
      <Collapsible.Main>
        <div>{'Main Section'}</div>
      </Collapsible.Main>
      <Collapsible.Content>
        <div>
          {
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vel erat sed purus hendrerit viverra. Duis et vestibulum metus. Sed consequat ut ante non vehicula. Etiam nunc massa, pharetra vel quam id, posuere rhoncus quam. Quisque imperdiet arcu enim, nec aliquet justo auctor eget. Curabitur in metus nec nunc rhoncus faucibus vitae ac elit. Nulla facilisi. Vestibulum quis pretium nulla. Nulla ut accumsan velit. Duis varius urna sed sem tempor, sit amet fermentum nibh auctor. Praesent lorem arcu, egestas non ante quis, placerat pellentesque lectus.Vestibulum lacinia ipsum quis venenatis tristique. Vivamus suscipit, libero eu fringilla egestas, orci urna commodo arcu, vel gravida turpis ipsum molestie nibh. Donec cursus eu ante sagittis ultrices. Phasellus id sagittis risus. Mauris dapibus neque faucibus, tempor ligula vel, cursus ante. Donec faucibus gravida porta. Nullam egestas est quis aliquam feugiat. Sed eget metus diam. Cras eget placerat libero.'
          }
        </div>
      </Collapsible.Content>
    </Collapsible>
  )

  const kit = screen.getByTestId(testId)
  expect(kit).toHaveClass(`${kitClass} additional_class`)
})
