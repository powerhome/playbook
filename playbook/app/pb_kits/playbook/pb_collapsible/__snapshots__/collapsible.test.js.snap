// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html structure is correct 1`] = `
<div>
  <div
    class="pb_collapsible_kit additional_class"
    data-testid="collapsible1"
  >
    <div
      class="pb_collapsible_main_kit cursor_pointer"
    >
      <div>
        <div
          class="pb_flex_kit_orientation_row_justify_content_left_align_items_center_spacing_between"
        >
          <div
            class="pb_flex_item_kit"
          >
            <div>
              Main Section
            </div>
          </div>
          <div
            class="pb_flex_item_kit"
          >
            <div
              class="icon_wrapper"
              style="vertical-align: middle; color: rgb(193, 205, 214);"
            >
              <i
                class="pb_icon_kit far fa-lg fa-fw fa-lg fa-chevron-down"
              />
              <span
                aria-label="chevron-down icon"
                hidden=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pb_collapsible_content_kit toggle-content"
      data-collapsible-content="true"
      style="height: 0px;"
    >
      <div>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vel erat sed purus hendrerit viverra. Duis et vestibulum metus. Sed consequat ut ante non vehicula. Etiam nunc massa, pharetra vel quam id, posuere rhoncus quam. Quisque imperdiet arcu enim, nec aliquet justo auctor eget. Curabitur in metus nec nunc rhoncus faucibus vitae ac elit. Nulla facilisi. Vestibulum quis pretium nulla. Nulla ut accumsan velit. Duis varius urna sed sem tempor, sit amet fermentum nibh auctor. Praesent lorem arcu, egestas non ante quis, placerat pellentesque lectus.Vestibulum lacinia ipsum quis venenatis tristique. Vivamus suscipit, libero eu fringilla egestas, orci urna commodo arcu, vel gravida turpis ipsum molestie nibh. Donec cursus eu ante sagittis ultrices. Phasellus id sagittis risus. Mauris dapibus neque faucibus, tempor ligula vel, cursus ante. Donec faucibus gravida porta. Nullam egestas est quis aliquam feugiat. Sed eget metus diam. Cras eget placerat libero.
      </div>
    </div>
  </div>
</div>
`;
